source/ui/dataman/AMScanThumbnailGridView.cpp

```diff
@@ -4,415 +4,443 @@
+    connect(inputManager_, SIGNAL(dragBegun()), this, SLOT(onDragBegun()));
+    connect(inputManager_, SIGNAL(selectionRectangleChanged(QRect)), this, SLOT(onSelectionRectangleChanged(QRect)));
+    connect(inputManager_, SIGNAL(selectionRectangleEnded(QRect, QItemSelectionModel::SelectionFlags)), this, SLOT(onSelectionRectangleEnded(QRect, QItemSelectionModel::SelectionFlags)));
+    connect(inputManager_, SIGNAL(hoverMove(int, int, int)), this, SLOT(onHoverMove(int, int, int)));
+
+    // Have to set mouse tracking in order to receive events when no mouse button is held
+    setMouseTracking(true);
+
+    setItemDelegate(new AMScanThumbnailGridViewItemDelegate(this));
+    horizontalScrollBar()->setVisible(false);
+    updateScrollBars();
+    rubberBand_ = new QRubberBand(QRubberBand::Rectangle, viewport());
+    rubberBand_->setVisible(false);
 }
 
 AMScanThumbnailGridView::~AMScanThumbnailGridView()
 {
-	delete geometryManager_;
-	inputManager_->deleteLater();
+    delete geometryManager_;
+    inputManager_->deleteLater();
 }
 
 QModelIndex AMScanThumbnailGridView::indexAt(const QPoint &point) const
 {
-	if(!model())
-		return QModelIndex();
+    if(!model())
+        return QModelIndex();
 
-	QPoint adjustedPoint(point.x() + horizontalOffset(),
-						 point.y() + verticalOffset());
+    QPoint adjustedPoint(point.x() + horizontalOffset(),
+                         point.y() + verticalOffset());
 
-	int integerIndexAt = geometryManager_->contentIndexAt(adjustedPoint);
+    int integerIndexAt = geometryManager_->contentIndexAt(adjustedPoint);
 
-	if(integerIndexAt == -1)
-		return QModelIndex();
+    if(integerIndexAt == -1)
+        return QModelIndex();
 
-	return model()->index(integerIndexAt, 1, QModelIndex());
+    return model()->index(integerIndexAt, 1, QModelIndex());
 }
 
 QRect AMScanThumbnailGridView::visualRect(const QModelIndex &index) const
 {
 
-	QRect rowContentRect;
-
-	if(index.parent().isValid()) {
-		// is  a thumbnail
-		rowContentRect = geometryManager_->contentGeometryAt(index.parent().row(),
-															 horizontalOffset(),
-															 verticalOffset());
-	} else {
-		rowContentRect = geometryManager_->contentGeometryAt(index.row(),
-															 horizontalOffset(),
-															 verticalOffset());
-	}
-
-	if(!index.parent().isValid()) {
-		// Is Scan
-		switch(index.column()) {
-		case 0:
-			return rowContentRect;
-		case 1:
-			return geometryManager_->scanNameRectangle(rowContentRect);
-		case 3:
-			return geometryManager_->scanDateRectangle(rowContentRect);
-
-		case 4:
-			return geometryManager_->scanTechniqueRectangle(rowContentRect);
-
-		default:
-			return QRect();
-		}
-	} else {
-		// Is Thumbnail
-		switch(index.column()) {
-		case 1:
-			return geometryManager_->thumbnailImageRectangle(rowContentRect);
-		case 2:
-			return geometryManager_->thumbnailTitleRectangle(rowContentRect);
-
-		default:
-			return QRect();
-		}
-	}
+    QRect rowContentRect;
+    /*
+    if(!index.isValid()){
```

Comment:
@iainworkman Yes this should be removed, it was a left over fragment from that first solution attempt. I removed it, tested, and pushed the update.